helper/call.go

```diff
@@ -68,21 +77,15 @@ type ContractCaller struct {
 	MainChainRPC     *rpc.Client
 	MaticChainClient *ethclient.Client
 
-	RootChainInstance     *rootchain.Rootchain
-	StakingInfoInstance   *stakinginfo.Stakinginfo
-	ValidatorSetInstance  *validatorset.Validatorset
-	StateSenderInstance   *statesender.Statesender
-	StateReceiverInstance *statereceiver.Statereceiver
-	StakeManagerInstance  *stakemanager.Stakemanager
-	MaticTokenInstance    *erc20.Erc20
-
 	RootChainABI     abi.ABI
 	StakingInfoABI   abi.ABI
 	ValidatorSetABI  abi.ABI
 	StateReceiverABI abi.ABI
 	StateSenderABI   abi.ABI
 	StakeManagerABI  abi.ABI
 	MaticTokenABI    abi.ABI
+
+	ContractInstanceCache *lru.Cache
```

Comment:
You don't need to use `lru.Cache`. Simple `map[string]interface{}` should work since it won't change very frequently.